layout/js/animate.js

```diff
@@ -8,43 +8,34 @@ var interval = undefined;
 var intervalLength = 160;
 var timestep = 1;
 
-var setColors = function() {
-  $.get( "js/precip-colors.json", function( data ) {
-    prcpColors = data
-  });
-  $.get( "js/times.json", function( data ) {
-    prcpTimes = data
-  });
-  svg = document.querySelector("svg");
-  pt = svg.createSVGPoint();
-}
+var fetchPrcpColors = $.get("js/precip-colors.json").done(function(data) {
+  prcpColors = data;
+});
+var fetchPrcpTimes = $.get("js/times.json").done(function(data) {
+  prcpTimes = data;
+});
 
 var animatePrcp = function(timestep) {
-  for (var i=0; i<prcpColors.length; i++) {
-    var bin = i + 1;
-    var selector = ".prcp-" + timestep + "-" + bin;
-    var stormDot = ".storm-dot";
-    var color = prcpColors[i];
-    // switch to style for transition
-    var style = {
-      "fill": color,
-      "transition": "fill 2s"
-    };
-    $(selector).css("fill", color);
-    $(stormDot).css("opacity", "0");
-    $(stormDot).css("transform", "scale(0.1)");
-    var storm = document.getElementById("storm-" + timestep);
-    if (!storm){
-    } else {
-      storm.setAttribute('style','opacity: 1.0;')
-      storm.setAttribute('transform','scale(1.0)')
+  prcpColors.forEach(function(color, index) {
+    var bin = index + 1;
+    var $prcpBin = $('.p-' + timestep + '-' + bin);
+    var $stormDot = $('.storm-dot');
+    var $currentStormDot = $('#storm-' + timestep)
+
+    $prcpBin.css("fill", color);
+    
+    $stormDot.css("opacity", "0").css("transform", "scale(0.1");
+    
+    if ($currentStormDot){
+      $currentStormDot.css('opacity', '1.0').css('transform', 'scale(0.1)');
     }
-  }
-  document.getElementById('timestamp-text').firstChild.data = prcpTimes.times[timestep-1]; // zero indexed
+  });
+  //Is this setting the html value
+  //document.getElementById('timestamp-text').firstChild.data = prcpTimes.times[timestep-1]; // zero indexed
   var darkWidth = timestep/prcpTimes.times.length;
-  document.getElementById('spark-light-mask').setAttribute('x', darkWidth);
-  document.getElementById('spark-light-mask').setAttribute('width', 1-darkWidth);
-  document.getElementById('spark-full-mask').setAttribute('width',darkWidth);
+  //document.getElementById('spark-light-mask').setAttribute('x', darkWidth);
```

Comment:
thanks - we can hook these up later?

Reply:
Yes. They throw errors right now. Although I could write this so they won't. I'll work on that.